packages/configure-splash-screen/src/android/index.ts

```diff
@@ -1,49 +1,28 @@
 import path from 'path';
 
-import { ResizeMode, Arguments, StatusBarOptions, AndroidOnlyStatusBarOptions } from '../constants';
 import configureAndroidManifestXml from './AndroidManifest.xml';
 import configureColorsXml from './Colors.xml';
 import configureDrawableXml from './Drawable.xml';
 import configureDrawables from './Drawables';
 import configureMainActivity from './MainActivity';
 import configureStylesXml from './Styles.xml';
+import { validateAndroidConfig } from '../validators';
+import { AndroidSplashScreenJsonConfig } from '../types';
 
 export default async function configureAndroid(
   projectRootPath: string,
-  {
-    resizeMode,
-    backgroundColor,
-    darkModeBackgroundColor,
-    imagePath,
-    darkModeImagePath,
-    statusBarHidden,
-    statusBarStyle,
-    darkModeStatusBarStyle,
-    statusBarTranslucent,
-    statusBarBackgroundColor,
-    darkModeStatusBarBackgroundColor,
-  }: Arguments &
-    Partial<StatusBarOptions> &
-    Partial<AndroidOnlyStatusBarOptions> & { resizeMode: ResizeMode }
+  configJSON: AndroidSplashScreenJsonConfig
 ) {
+  const validatedConfig = await validateAndroidConfig(configJSON);
+
   const androidMainPath = path.resolve(projectRootPath, 'android/app/src/main');
 
   await Promise.all([
-    configureDrawables(androidMainPath, imagePath, darkModeImagePath),
-    configureColorsXml(androidMainPath, {
-      backgroundColor,
-      darkModeBackgroundColor,
-      statusBarBackgroundColor,
-      darkModeStatusBarBackgroundColor,
-    }),
-    configureDrawableXml(androidMainPath, resizeMode),
-    configureStylesXml(androidMainPath, {
-      statusBarHidden,
-      statusBarStyle,
-      darkModeStatusBarStyle,
-      addStatusBarBackgroundColor: !!statusBarBackgroundColor,
-    }),
+    configureDrawables(androidMainPath, validatedConfig),
```

Comment:
Is there a particular reason for this change? I quite liked the idea of `configureX` functions as small, isolated, reusable commands which aligned more with the idea of passing only the required information to the method instead of the whole config object (it also makes it easier to write more tangled code). Just curious!

Reply:
Ease of writing the code I think 🤔 
Instead of creating massive deconstruction od the config object and aliasing nested properties just to pass them to other function invocation I have a single point of reference in each file - the config object. I see that this is sth that might confuse the reader and I dislike it, so it should be refactored like you're suggesting 👍

Reply:
I've adjusted it a little. These functions are now expecting only parameters that they're consuming, but TS still allows us to pass some `bigger` object that satisfies the `smaller`/`subset` type 😉